### DISCUSSION TITLE
Module 'FirebaseCore' not found

### DISCUSSION THREAD

**Discussion by saoudnazir** [invertase/react-native-firebase#6339] [Q&A] [answered]
I am facing error when integrating Firebase Authentication on iOS project. 
<img width="1006" alt="Screen Shot 2022-06-26 at 5 29 43 pm" src="https://user-images.githubusercontent.com/31433177/175804190-471c8494-9180-44c3-b013-29cf54a0392d.png">

But when I change "@import FirebaseCore;" to "#import "FirebaseCore/FirebaseCore.h". everything feels working.
File location: Pods/FirebaseAuth/FirebaseCoreInternal.h

Please help me where I am wrong.

Thanks

**Comment by mikehardy** [comment]
Hi there!
This is an issue not a discussion. And the reason we use issues is because we have a template there where you would provide the information we need to help you.

You provided none of that unfortunately, so we can't help you. https://stackoverflow.com/help/how-to-ask

Additionally in that template it asks the question if you have searched for similar issues. My guess is this is covered in one of the recent issues with regard to react-native-firebase v15, which I can only guess you are using since you did not log an issue so you did not fill in version details so I can only guess.

You will probably need to make sure you have followed install instructions here https://rnfirebase.io/#altering-cocoapods-to-use-frameworks

I can only guess at your setup, so if you are using Expo, note that we try to help Expo users but we do not use Expo ourselves as maintainers, so we rely on the community to help, and currently we do not have a 100% answer for how to do the things you need for Expo and react-native-firebase v15, but users have reported they got it running here 

https://github.com/invertase/react-native-firebase/issues/6332#issuecomment-1166190120

**Reply by aabsharkhan1** [reply]
i am facing the same issue please look into it just followed the doument's instructions ..

**Comment by linhnguyen2307** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/86759727/182308808-f77e8fd8-5275-4404-9d45-318477c08ad8.png)
hi , In my case I just add this line + pod install and it build succeed :
 pod 'FirebaseCore', :modular_headers => true

**Reply by hongjie104** [reply]
it works for me!

**Reply by fukemy** [reply]
love u

**Reply by mikehardy** [reply]
This is a dead-end solution though - I see it has lots of attention but everyone *must* understand that avoiding use of `use_frameworks! :linkage => :static` in your Podfile is a workaround with no future. It is not supported in general and will specifically cause errors if you use storage and functions modules already, more in the future. This is coming from the underlying firebase-ios-sdk and you *must* start adapting.

**Reply by Michae-zHOU** [reply]
work like charm!!!

**Reply by ghost** [reply]
From Vietnam with love

**Reply by Zohaibmalikdev** [reply]
Thanks with Love <3

**Reply by jaredly** [reply]
fwiw, I don't need storage or functions or almost all of firebase, I only need Messaging, and I'd hate to give up flipper AND fabric just for compatibility with some libraries that I won't even use.
Additionally, and it's possible [I'm reading these docs wrong](https://github.com/firebase/firebase-ios-sdk/blob/master/docs/firebase_in_libraries.md), but it looks like using firebase-ios-sdk non-statically *is* supported, starting with 7.x. And because I'm referencing it directly in my app, and not from a library, it doesn't seem like the stated pitfalls apply?

**Reply by mikehardy** [reply]
If I understand correctly, as the firebase-ios-sdk team continues converting to Swift internally, the techniques they used in functions and storage that make building it without use_frameworks unsupportable will be applied elsewhere. There is limited ability to support a lot of build variations here when the issue/discussion load related to build errors is considered in aggregate (that is: lots of people have limited experience with solving native build problems and they post issues here requesting help...) so we support one style that works consistently.

You're free to disregard any requirement in an open source context, that's one of the beauties of open source, but it is not reasonable to expect support if requirements are disregarded. That seems like an even stance.

**Reply by erdemalptugoksuz** [reply]
**thx it's work!**

**Reply by duongnguyen17** [reply]
love uuuuuuuu

**Reply by Abramovick** [reply]
@jaredly Did you manage to get it working without losing flipper & fabric?

**Comment by deggertsen** [comment]
What worked for me, just in case it helps anyone else.
1. Do as @mikehardy says and fork and follow the readme instructions to build https://github.com/mikehardy/rnfbdemo
2. Compare the Podfile and AppDelegate.mm files with what you have and make adjustments. This is the hardest part if you don't understand what everything in the Podfile is doing.
3. Remove package dependencies in Xcode https://github.com/invertase/react-native-firebase/issues/6304#issuecomment-1273762869

**Comment by sirTangale** [comment]
in FirebaseCoreInternal.h
@import FirebaseCore;"  ->  "#import "FirebaseCore/FirebaseCore.h".

and in podfile  
pod 'GoogleUtilities', :modular_headers => true

**Reply by mikehardy** [reply]
If you use these modular_headers workarounds, there will be no access to support here or in firebase-ios-sdk repository when you have problems. This is not a supported configuration.

**Comment by hemanthAppmaker** [comment]
the fix was to add this line below  **target ['your app'] do** in pod file

pod 'FirebaseCore', :modular_headers => true



  [1]: https://github.com/invertase/react-native-firebase/discussions/6339#discussioncomment-3304918

**Reply by mikehardy** [reply]
> pod 'FirebaseCore', :modular_headers => true

No, that's not the fix, that's an unsupported workaround. If you need support in this repo and your Podfile contains that vs `use_frameworks! :linkage => :static`, you will not be able to access support.

**Reply by VictorHugo110199** [reply]
This is a little strange, as Google's own Firebase installation document asks for this change to be made.

**Reply by mikehardy** [reply]
Please reference the documentation with a URL @VictorHugo110199 - specifically as they relate to use of Firebase in *source-based* contexts (as technically + programmatically - even though you may think of react-native-firebase as a library all react-native modules are actually just chunks of source that you include in your code but appear as "just more code" to the build system, not as libraries, e.g. xcframeworks etc)

**Reply by VictorHugo110199** [reply]
Here is the link to the Google documentation: https://rnfirebase.io/#3-react-native-cli---ios-setup

I took the liberty of taking some screenshots to make it easier to understand.
![sentiu1](https://github.com/invertase/react-native-firebase/assets/106779643/afe3b9ae-b430-45c8-97e4-5cc02e879e93)

**Reply by mikehardy** [reply]
That's self-referential, that's not the google docs (those would be on a google.com domain) those are *our* docs for this module here, note "rnfirebase.io" as the domain

Also, they indicate what I said: use_frameworks is a strict requirement.

So that does not seem to advance the discussion. I am not aware of documentation on a google.com domain that says when using firebase-ios-sdk in a source-based context you can do anything other than use_frameworks

**Comment by iamd2epak** [comment]
pod 'GoogleUtilities', :modular_headers => true;
  pod 'FirebaseCore', :modular_headers => true
  
  
adding these lines in my podfile fixed it. add these line just below 
target 'MyAppName' do
  use_expo_modules!
